### ISSUE TITLE
2008 R2 Memory Dump

### ISSUE THREAD

**Issue by TheManGlan** [trimble-oss/dba-dash#972]
Hi, 
We are experiencing memory dumps on monitored sql servers of version 10.50.x (2008 R2). 
It causes the instance to break down, and it needs to be restarted. 
Has anyone else experienced this? 
It happens within an hour after adding the sql servers in the configurator.

**Comment by Drakokorian** [comment]
If your seeing that check your memory clerk XE usage on that server Extended events are stealing all the memory it is a bug in SQL server 2016 and all the ways down: https://github.com/trimble-oss/dba-dash/issues/951

**Comment by TheManGlan** [comment]
> If your seeing that check your memory clerk XE usage on that server Extended events are stealing all the memory it is a bug in SQL server 2016 and all the ways down: #951

We haven't even tested that yet. So in our test environment, when adding new SQL Servers, those settings are not enabled. 
![image](https://github.com/user-attachments/assets/aaccd1ea-e10c-48df-b8c0-4ed0bcf9a1fc)

**Comment by Drakokorian** [comment]
I have not seen that in 2008 for me

**Comment by TheManGlan** [comment]
> I have not seen that in 2008 for me

It only happens on 2008 **R2**. We have tested it on several 2008 instances without any issues.

**Comment by DavidWiseman** [comment]
You might find some more details about the dump in the error log and the Logs folder.  The dump will also be located in the logs folder.  Usually I'd recommend contacting MS support for help with diagnosis, but SQL 2008 R2 is out of support.  Hopefully there are some clues in the error log.  You can do some analysis of the dump file yourself with WinDbg.  There is also an [online dump analysis tool](https://www.sqldumpanalysis.com/) you could try.  

I have a SQL 2008 R2 instance in my lab and I don't experience this issue.  If you are not on the latest patch version (SP3) it might be worth upgrading.  SQL 2008 R2 won't get any further patches.

Most of the DBA Dash collections are just querying the DMVs - nothing dangerous or unusual.  If running these are triggering a dump it's likely you've hit a SQL Server bug.  Hopefully you can identify what is causing the issue.  As a last resort you could use a process of elimination by selectively enabling and disabling collections. [See here.](https://dbadash.com/docs/help/schedule/)

Please post back with anything you find.  Thanks.

**Comment by DavidWiseman** [comment]
Are you using database mirroring?  It might be [this collection](https://raw.githubusercontent.com/trimble-oss/dba-dash/main/DBADash/SQL/SQLCorruption.sql) causing the problem if you also have databases in a restoring state. 
It can also occur without database mirroring if you have an old version of DBA Dash installed (2.45 and lower)
#682 

If it is this issue there are a couple of workarounds:
* Disable the collection
* Use a sysadmin user to collect the data.

**Comment by TheManGlan** [comment]
> You might find some more details about the dump in the error log and the Logs folder. The dump will also be located in the logs folder. Usually I'd recommend contacting MS support for help with diagnosis, but SQL 2008 R2 is out of support. Hopefully there are some clues in the error log. You can do some analysis of the dump file yourself with WinDbg. There is also an [online dump analysis tool](https://www.sqldumpanalysis.com/) you could try.
> 
> I have a SQL 2008 R2 instance in my lab and I don't experience this issue. If you are not on the latest patch version (SP3) it might be worth upgrading. SQL 2008 R2 won't get any further patches.
> 
> Most of the DBA Dash collections are just querying the DMVs - nothing dangerous or unusual. If running these are triggering a dump it's likely you've hit a SQL Server bug. Hopefully you can identify what is causing the issue. As a last resort you could use a process of elimination by selectively enabling and disabling collections. [See here.](https://dbadash.com/docs/help/schedule/)
> 
> Please post back with anything you find. Thanks.

Hi David, 

Dump files indicate that an EXCEPTION_ACCESS_VIOLATION occurs.
I admit that we are testing it on old versions that we shouldn't be using anymore—but you know the story with that issue.
We are testing it on SP1 but should definitely test it on SP3. Regarding the upgrade, it might be just as easy to upgrade the major versions instead of the SP versions since production downtime needs to be planned anyway.
I am working on the elimination process now and have disabled these components for the time being, but without success. I will continue! :)

AzureDBServiceObjectives
AzureDBResourceGovernance
AzureDBResourceStats
AzureDBElasticPoolResourceStats
PerformanceCounters
AvailabilityGroups
AvailabilityReplicas
DriversWMI
IOStats

Regarding mirroring, we have tested on both mirrored and non-mirrored servers.
There was no difference, so I don't suspect this to be the root cause.

Thank you for your ideas! I will get back to you with any test results as soon as possible.

**Comment by DavidWiseman** [comment]
The [Corruption collection](https://raw.githubusercontent.com/trimble-oss/dba-dash/main/DBADash/SQL/SQLCorruption.sql) could be worth ruling out as that's the collection that had the issue previously.  The issue is a bug accessing **sys.dm_db_mirroring_auto_page_repair** if the user isn't sysadmin and there are also databases in a RESTORING state.  If you are using an old version of DBA Dash, this might trigger regardless if mirroring is enabled.

A common cause of bugs in SQL Server is AV software loading itself into SQL Server address space.  DBA Dash has a OS Loaded Modules tab under the Checks node you can use to check for this.  Or you can query [sys.dm_os_loaded_modules](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-os-loaded-modules-transact-sql).  There is some advice on configuring AV for SQL Server [here](https://learn.microsoft.com/en-us/troubleshoot/sql/database-engine/security/antivirus-and-sql-server).  It might not have anything to do with this but it's worth a check.

**Comment by TheManGlan** [comment]
> The [Corruption collection](https://raw.githubusercontent.com/trimble-oss/dba-dash/main/DBADash/SQL/SQLCorruption.sql) could be worth ruling out as that's the collection that had the issue previously. The issue is a bug accessing **sys.dm_db_mirroring_auto_page_repair** if the user isn't sysadmin and there are also databases in a RESTORING state. If you are using an old version of DBA Dash, this might trigger regardless if mirroring is enabled.
> 
> A common cause of bugs in SQL Server is AV software loading itself into SQL Server address space. DBA Dash has a OS Loaded Modules tab under the Checks node you can use to check for this. Or you can query [sys.dm_os_loaded_modules](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-os-loaded-modules-transact-sql). There is some advice on configuring AV for SQL Server [here](https://learn.microsoft.com/en-us/troubleshoot/sql/database-engine/security/antivirus-and-sql-server). It might not have anything to do with this but it's worth a check.

In our case, we have granted the user sysadmin from the beginning, and the intention is to lower down the permissions based upon your documention during later test phases. And as written before, we are testing on the latest version of DBA Dash. But for sure, it makes sense to mention it to others who might experience the same issues. 

I can see that our AV is in fact a loaded module on the instances having the issue - So we will look into this also.

**Comment by TheManGlan** [comment]
**Update:** 
Root Cause: The [Corruption collection](https://raw.githubusercontent.com/trimble-oss/dba-dash/main/DBADash/SQL/SQLCorruption.sql). 
The user we are using in the test scenario is sysadmin. 
Databases are not in a restoring state before or after the memory dump. 
We are using the latest DBA Dash. 
We have testet it on both mirrored and non-mirrored SQL Server 2008 R2 SP1 instances.

**Comment by TheManGlan** [comment]
Just a quick update on our tests:

We've ruled out the AV as the cause since, even after uninstalling it, the SQL Server service still stopped after a certain period. I also tried disabling all collectors and then re-enabling them one by one. Each collector was allowed to run for several hours, and it wasn't until all the "1-minute" and "Every Hour" collectors were enabled that the service failed again.

During the tests, I monitored memory utilization, which peaked at 30-40% when the collectors were running and then quickly dropped back down.

The only remaining step might be to upgrade to SP3.

**Comment by DavidWiseman** [comment]
Thank you for the update.  I did a quick check with a RTM version of SQL 2008 R2 and couldn't reproduce the issue.  I'll try again with SP1 when I get chance.  There could be a very specific set of circumstances that trigger the issue though.

**Comment by TheManGlan** [comment]
> Thank you for the update. I did a quick check with a RTM version of SQL 2008 R2 and couldn't reproduce the issue. I'll try again with SP1 when I get chance. There could be a very specific set of circumstances that trigger the issue though.

I believe the issue is related specifically to 2008 R2 SP1. We upgraded it yesterday afternoon, and it has been running smoothly since. Normally, as I mentioned, it would crash within hours.

Our tests indicate that the problem is indeed connected to SP1, regardless of hardware resources. We tested it on two different 2008 R2 SP1 servers with different hardware configurations, and both crashed within hours.

But SP3 seems promising.
